dejavu_bot.py

```diff
@@ -1,111 +1,152 @@
             break
 
+
 def get_rand_datetime(start, end):
     """
     https://stackoverflow.com/questions/553303/generate-a-random-date-between-two-other-dates
 
-    This function will return a random datetime between two datetime 
-    objects.
+    This function will return a random datetime between two datetime objects.
     """
     delta = end - start
     int_delta = (delta.days * 24 * 60 * 60) + delta.seconds
     random_second = randrange(int_delta)
     return start + timedelta(seconds=random_second)
 
-async def create_and_send_response(rand_message, channel, arg):
-    """
-    Creates an image with the message author, content, and creation datetime
-    with a random colored background
-    """
+
+async def create_and_send_response(rand_message, channel, choice):
+
     text = (
-        rand_message.author.name +
-        " said: \n" +
-        rand_message.content +
-        "\nat " +
-        rand_message.created_at.strftime("%Y-%m-%d %I:%M %p")
+        rand_message.author.name
+        + " said: \n"
+        + rand_message.content
+        + "\nat "
+        + rand_message.created_at.strftime("%Y-%m-%d %I:%M %p")
     )
 
-    if arg == 'image':
+    if choice == "text":
+        # Sends the random message with the message author, content,
+        # and creation datetime
+        # `/dejavu text`
+        await channel.send(text)
+    elif choice == "image":
         await create_and_send_image(text, channel)
-    elif arg == 'whosaid':
-        # if the arg is whosaid, store the message details in the bot's who_said_context
-        bot.who_said_context = {
-            'author_id': rand_message.author.id,
-            'content': rand_message.content,
-            'channel': channel
-        }
-        await who_said(rand_message.content, channel)
+    elif choice == "whosaid":
+        # if the choice is whosaid, pass the rand_message and channel to whosaid()
+        await whosaid(rand_message, channel)
     else:
-        # /dejavu text
-        # Just return the random message as text
-        await channel.send(text)
+        await channel.send("Invalid Command.")
+
 
 async def create_and_send_image(text, channel):
     """
-    Handle the case where an image is requested
-    /dejavu image
+    Creates an image with the message author, content, and creation datetime
+    with a random colored background. TODO: Doesn't work right.
+    `/dejavu image`
     """
-    font = ImageFont.truetype(
-        "./fonts/Courier.ttf", # debian path
-        size=14
-    )
+    font = ImageFont.truetype("./fonts/Courier.ttf", size=14)  # debian path
 
     # Convert dict_items to a list
     color_items = list(ImageColor.colormap.items())
-    rand_color = choice(color_items)[0]
-    img = Image.new('RGB', (1000, 100), color=rand_color)
+
+    rand_color_name = choices(list(ImageColor.colormap.keys()))[0]
+    rand_color = ImageColor.getrgb(rand_color_name)
+    img = Image.new("RGB", (1000, 100), color=rand_color)
```

Comment:
I feel like this was more readable before. Can we change this back to using `choice`? I think this function should still work if we leave it alone.

Reply:
:+1:

Reply:
:+1: